pkg/search/postgres/store_cache.go

```diff
@@ -321,6 +321,9 @@ func (c *cachedStore[T, PT]) GetMany(ctx context.Context, identifiers []string)
 // WalkByQuery iterates over all the objects scoped by the query applies the closure.
 func (c *cachedStore[T, PT]) WalkByQuery(ctx context.Context, query *v1.Query, fn func(obj PT) error) error {
 	defer c.setCacheOperationDurationTime(time.Now(), ops.WalkByQuery)
+	if query == nil || query.EqualVT(search.EmptyQuery()) {
```

Comment:
Do we need something like in Count?

https://github.com/stackrox/stackrox/blob/4623734ae5999fc4588e27212aabb8da16c51dd4/pkg/search/postgres/store_cache.go#L249-L257